stackdriver/storage.go

```diff
@@ -72,7 +73,30 @@ func (s *Storage) Close() error {
 // ApplyConfig updates the state as the new config requires.
 func (s *Storage) ApplyConfig(conf *config.Config) error {
 	var projectId string
-	if value, ok := conf.GlobalConfig.ExternalLabels[ProjectIdLabel]; !ok {
+	var location string
+	var clusterName string
+	el := conf.GlobalConfig.ExternalLabels
+	if md.OnGCE() {
```

Comment:
Neat. We will need to update the public documentation at https://cloud.google.com/monitoring/kubernetes-engine/prometheus#prometheus. How would you describe this feature to a user going through step #5 of those instructions?

Reply:
Hmm, yeah, I think the feature could be introduced as "if you run GKE, you can leave project metadata blank", which I think would not help to reduce user burden and could possibly cause confusion. I suggest we just hide this from user?